crates/ui/src/windows/system.rs

```diff
@@ -0,0 +1,680 @@
+            ),
+            title_modal: LabelModal::new("title_modal".into(), "Graphics/Titles".into()),
+            gameover_modal: LabelModal::new("gameover_modal".into(), "Graphics/Gameovers".into()),
+            battle_transition_modal: LabelModal::new(
+                "battle_transition_modal".into(),
+                "Graphics/Transitions".into(),
+            ),
+            battle_bgm_modal: SoundModal::new(luminol_audio::Source::BGM, "battle_bgm_modal"),
+            battle_end_me_modal: SoundModal::new(luminol_audio::Source::ME, "battle_end_me_modal"),
+            gameover_me_modal: SoundModal::new(luminol_audio::Source::ME, "gameover_me_modal"),
+            cursor_se_modal: SoundModal::new(luminol_audio::Source::SE, "cursor_se_modal"),
+            decision_se_modal: SoundModal::new(luminol_audio::Source::SE, "decision_se_modal"),
+            cancel_se_modal: SoundModal::new(luminol_audio::Source::SE, "cancel_se_modal"),
+            buzzer_se_modal: SoundModal::new(luminol_audio::Source::SE, "buzzer_se_modal"),
+            equip_se_modal: SoundModal::new(luminol_audio::Source::SE, "equip_se_modal"),
+            shop_se_modal: SoundModal::new(luminol_audio::Source::SE, "shop_se_modal"),
+            save_se_modal: SoundModal::new(luminol_audio::Source::SE, "save_se_modal"),
+            load_se_modal: SoundModal::new(luminol_audio::Source::SE, "load_se_modal"),
+            battle_start_se_modal: SoundModal::new(
+                luminol_audio::Source::SE,
+                "battle_start_se_modal",
+            ),
+            escape_se_modal: SoundModal::new(luminol_audio::Source::SE, "escape_se_modal"),
+            actor_collapse_se_modal: SoundModal::new(
+                luminol_audio::Source::SE,
+                "actor_collapse_se_modal",
+            ),
+            enemy_collapse_se_modal: SoundModal::new(
+                luminol_audio::Source::SE,
+                "enemy_collapse_se_modal",
+            ),
+
+            max_elements: None,
+        }
+    }
+}
+
+impl luminol_core::Window for Window {
+    fn id(&self) -> egui::Id {
+        egui::Id::new("system_editor")
+    }
+
+    fn requires_filesystem(&self) -> bool {
+        true
+    }
+
+    fn show(
+        &mut self,
+        ctx: &egui::Context,
+        open: &mut bool,
+        update_state: &mut luminol_core::UpdateState<'_>,
+    ) {
+        let data = std::mem::take(update_state.data); // take data to avoid borrow checker issues
+        let mut system = data.system();
+        let actors = data.actors();
+
+        let mut modified = false;
+
+        if self.max_elements.is_none() {
+            self.max_elements = Some(system.elements.len().saturating_sub(1));
+        }
+
+        egui::Window::new("System Editor")
+            .id(self.id())
+            .default_width(500.)
+            .open(open)
+            .show(ctx, |ui| {
+                ui.with_cross_justify(|ui| {
+                    egui::ScrollArea::vertical().show(ui, |ui| {
```

Comment:
These pickers should probably be in a grid layout
![image](https://github.com/user-attachments/assets/8f45d6fb-f764-42bd-8e3d-f84977cdde14)